### ISSUE TITLE
proof is in the pudding - go playground for examples

### ISSUE THREAD

**Issue by headwinds** [samber/lo#226]
first off, fantastic library! this is more praise and wanting to help myself and others learn than an issue.

Has already started to build a collection of examples using the go playground? 

I was struggling a bit to learn how to use filter against a collection so [created this playground] (https://go.dev/play/p/dxXmzkdCRPw) to figure it out.

**Comment by samber** [comment]
hi @headwinds 

I made a small PR: #230

Can you review it, please?

As soon as we are ok with this first PR, we can distribute work ;)

**Comment by headwinds** [comment]
@samber I've reviewed & much approve - that's an excellent approach and should encourage more devs to contribute examples!

**Comment by samber** [comment]
![](https://media4.giphy.com/media/nFFguNjdeotwc/giphy.gif?cid=ecf05e47izczrehiy5acbqz2l2wtiyg3c1sfj5nou7ul6ytx&rid=giphy.gif&ct=g)

Can I ask you to write some examples for find.go ? I will keep pushing for slice.go and map.go.

I liked the example you provided above, but don't spend too much time with such big samples! ;)

I gonna write some instructions to future contributors in a PR template.

**Comment by samber** [comment]
I just added:
- condition_example_test.go
- map_example_test.go
- math_example_test.go
- slice_example_test.go
- string_example_test.go

**Comment by headwinds** [comment]
cool I'll take a look at your samples and could write one similar for find.go - keep it small simple - 😘

**Comment by headwinds** [comment]
[find playground](https://go.dev/play/p/X61OOFg-fp3) 

or maybe not so simple but I always look for examples that have both a simple string and an object

**Comment by headwinds** [comment]
[find playground that handles no results](https://go.dev/play/p/SMIOYGiG8aD)

**Comment by samber** [comment]
Can I suggest some changes ? 

-> https://go.dev/play/p/c2lDl0vxr-V

* i separated your examples into 2 functions to make it clear (go playground editor sucks)
* I added a check for `ok` 

In your last snippet, I'm not sure declaring `ProductSearchResult` is necessary. It makes the example longer to understand. You can just return `(Product, error)` instead.

**Comment by headwinds** [comment]
@samber Yes this is great - let's go with your version - I'm pretty new to go and can see the experience in your upgrades!

**Comment by headwinds** [comment]
Actually one small change - can we flip the function names and order - so that string example runs first then the struct example - simple before complex 

Currently it prints:

```
We also found the product Dragon Fruit ($3.75) 
We found the coconut
```

So I'd like to read: 

`````
We found the coconut
We also found the product Dragon Fruit ($3.75) 
````

And ship it!

**Comment by samber** [comment]
Still a few examples missing:
- channel_example_test.go
- concurrency_example_test.go
- find_example_test.go
- func_example_test.go
- intersect_example_test.go
- type_manipulation_example_test.go
